Make check_is_size clamp to sys.maxsize - 1, so sys.maxsize comparison returns False

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #122372
* #122370

Partially fixes https://github.com/pytorch/pytorch/issues/113002

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang

ezyang committed:
Update

[ghstack-poisoned]

lezcano reviewed (approved):
We should improve our value range comparisons within `subset` and the like by always clamping, as if you do `2 * x` for this value range we'll still have the same issue.

This is strictly better than the previous state, so lgtm

ezyang committed:
Update on "Make check_is_size clamp to sys.maxsize - 1, so sys.maxsize comparison returns False"


Partially fixes https://github.com/pytorch/pytorch/issues/113002

Signed-off-by: Edward Z. Yang <ezyangmeta.com>

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang

[ghstack-poisoned]

ezyang committed:
Update

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge
